### PR TITLE
SI-7251, compiler crash with $.

### PR DESCRIPTION
**Pull request by paulp**
We don't need to assert our way out of tight spots, we can
issue an error.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala [hunk 1]
```diff
@@ -1183,27 +1188,17 @@ abstract class ClassfileParser {
...
-            s
-
-          case None =>
-            classNameToSymbol(externalName)
+            val sym       = classSymbol(outerName)
+
+            enteringTyperIfPossible(getMember(sym, innerName.toTypeName)) orElse {
+              globalError(
+                s"""|Unable to load $externalName. Most commonly, this arises from referencing an object's
+                    |implementation-internal name instead of its name as defined within the scala language.""".stripMargin
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala]: you can `sm"""...."""` this now.

**Comment by paulp** [comment]
What the... hey @xeno-by, I changed an assertion to issue a regular error and suddenly it's failure city.

```
[scalacfork] error: Unable to load scala.tools.nsc.typechecker.Analyzer$namerFactory$. Most commonly, this arises from referencing an object's
[scalacfork] implementation-internal name instead of its name as defined within the scala language.
[scalacfork] error: Unable to load scala.tools.nsc.typechecker.Analyzer$packageObjects$. Most commonly, this arises from referencing an object's
[scalacfork] implementation-internal name instead of its name as defined within the scala language.
[scalacfork] error: Unable to load scala.tools.nsc.typechecker.Analyzer$typerFactory$. Most commonly, this arises from referencing an object's
[scalacfork] implementation-internal name instead of its name as defined within the scala language.
[scalacfork] error: Unable to load scala.tools.nsc.typechecker.StdAttachments$MacroRuntimeAttachment$. Most commonly, this arises from referencing an object's
```

This suggests you are catching the AssertionError (intentionally or otherwise) somewhere. Can you please a) not do that b) look into this?

**Comment by xeno-by** [comment]
The only thing that comes to mind is this: https://issues.scala-lang.org/browse/SI-6748. Also why do you think it's me? I mean, that's probably me, but could you share the information that has led you to this conclusion?

**Comment by paulp** [comment]
Sorry, the assumption happened on such a deeply unconscious level that I didn't even realize I was making an assumption. Let's call it "programmer's intuition" and if I'm wrong I'll buy you dinner.

**Comment by xeno-by** [comment]
How do I reproduce this failure? Checking out your branch and recompiling doesn't crash. Moreover, I cannot see the `quick.repl` task there. upd. Nevermind: `quick.repl` is there, but I still can't explain why it works for me and doesn't work for Jenkins.

```
Last login: Thu Mar 14 16:13:15 on ttys001
16:40 ~$ kep7251
16:40 ~/Projects/Kepler_7251 (ticket/7251)$ git remote add paulp git@github.com:paulp/scala.git
16:40 ~/Projects/Kepler_7251 (ticket/7251)$ gru
Fetching origin
Fetching upstream
Fetching paulp
remote: Counting objects: 8633, done.
remote: Compressing objects: 100% (3280/3280), done.
remote: Total 6996 (delta 4756), reused 5076 (delta 3238)
Receiving objects: 100% (6996/6996), 2.41 MiB | 362 KiB/s, done.
Resolving deltas: 100% (4756/4756), completed with 744 local objects.
From github.com:paulp/scala
 * [new branch]            add-synthetics-once -> paulp/add-synthetics-once
 * [new branch]            collections-extend-any -> paulp/collections-extend-any
 * [new branch]            foo-is-this-dot-foo -> paulp/foo-is-this-dot-foo
 * [new branch]            issue/3452 -> paulp/issue/3452
 * [new branch]            issue/4012 -> paulp/issue/4012
 * [new branch]            issue/5031 -> paulp/issue/5031
 * [new branch]            issue/5330-2 -> paulp/issue/5330-2
 * [new branch]            issue/6153 -> paulp/issue/6153
 * [new branch]            issue/6245 -> paulp/issue/6245
 * [new branch]            issue/6609 -> paulp/issue/6609
 * [new branch]            issue/6662 -> paulp/issue/6662
 * [new branch]            issue/6685 -> paulp/issue/6685
 * [new branch]            issue/7251 -> paulp/issue/7251
 * [new branch]            issue/785  -> paulp/issue/785
 * [new branch]            master     -> paulp/master
 * [new branch]            no-useless-conversions -> paulp/no-useless-conversions
 * [new branch]            numeric-range -> paulp/numeric-range
 * [new branch]            pr/asm-verify -> paulp/pr/asm-verify
 * [new branch]            pr/buildxml -> paulp/pr/buildxml
 * [new branch]            pr/eta-expansion-order -> paulp/pr/eta-expansion-order
 * [new branch]            pr/notNull -> paulp/pr/notNull
 * [new branch]            pr/warnings -> paulp/pr/warnings
 * [new branch]            prime      -> paulp/prime
 * [new branch]            topic/array-manifest -> paulp/topic/array-manifest
 * [new branch]            topic/classpath -> paulp/topic/classpath
 * [new branch]            topic/collection-creation -> paulp/topic/collection-creation
 * [new branch]            topic/contexterrors-events -> paulp/topic/contexterrors-events
 * [new branch]            topic/contrarivariance -> paulp/topic/contrarivariance
 * [new branch]            topic/ensure-fully-defined -> paulp/topic/ensure-fully-defined
 * [new branch]            topic/genjvm-java-source -> paulp/topic/genjvm-java-source
 * [new branch]            topic/global-assist -> paulp/topic/global-assist
 * [new branch]            topic/invert -> paulp/topic/invert
 * [new branch]            topic/literal-dot-type -> paulp/topic/literal-dot-type
 * [new branch]            topic/partest-in-bash -> paulp/topic/partest-in-bash
 * [new branch]            topic/partest-sprint -> paulp/topic/partest-sprint
 * [new branch]            topic/plausibly-inhabited -> paulp/topic/plausibly-inhabited
 * [new branch]            topic/positions -> paulp/topic/positions
 * [new branch]            topic/postConstructor -> paulp/topic/postConstructor
 * [new branch]            topic/purity-analysis -> paulp/topic/purity-analysis
 * [new branch]            topic/scalacheck -> paulp/topic/scalacheck
 * [new branch]            topic/specialized-small-seqs -> paulp/topic/specialized-small-seqs
 * [new branch]            topic/tree-traversers -> paulp/topic/tree-traversers
 * [new branch]            topic/unchecked -> paulp/topic/unchecked
 * [new branch]            topic/virtualized-pmat -> paulp/topic/virtualized-pmat
 * [new branch]            typer-timings -> paulp/typer-timings
 * [new branch]            wip/jline  -> paulp/wip/jline
 * [new branch]            wip/predicates -> paulp/wip/predicates
 * [new branch]            wip/warn-val-override -> paulp/wip/warn-val-override
 * [new branch]            zinc-build-system -> paulp/zinc-build-system
16:40 ~/Projects/Kepler_7251 (ticket/7251)$ gch paulp/issue/7251
Note: checking out 'paulp/issue/7251'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at d229fb5273... SI-7251, compiler crash with $.
16:41 ~/Projects/Kepler_7251 (HEAD)$ gch -b topic/7251
Switched to a new branch 'topic/7251'
16:41 ~/Projects/Kepler_7251 (topic/7251)$ ant all.clean build
Buildfile: /Users/xeno_by/Projects/Kepler_7251/build.xml

strap.clean:

pack.clean:
   [delete] Deleting directory /Users/xeno_by/Projects/Kepler_7251/build/pack

init.jars.check:

init.jars:
     [echo] Updating bootstrap libs.  (To do this by hand, run ./pull-binary-libs.sh)
     [exec] Resolving [943cd5c8802b2a3a64a010efb86ec19bac142e40/lib/ant/ant-contrib.jar]
     [exec] Resolving [3fc1e35ca8c991fc3488548f7a276bd9053c179d/lib/ant/ant-dotnet-1.0.jar]
     [exec] Resolving [7b456ca6b93900f96e58cc8371f03d90a9c1c8d1/lib/ant/ant.jar]
     [exec] Resolving [7e50e3e227d834695f1e0bf018a7326e06ee4c86/lib/ant/maven-ant-tasks-2.1.1.jar]
     [exec] Resolving [2c61d6e9a912b3253194d5d6d3e1db7e2545ac4b/lib/ant/vizant.jar]
     [exec] Resolving [f93a2525b5616d3a4bee7848fabbb2856b56f653/lib/forkjoin.jar]
     [exec] Resolving [a5261e70728c1847639e2b47d953441d0b217bcb/lib/jline.jar]
     [exec] Resolving [f9f41fb909df6a0178906c9fd02e5d0efa15c9ed/lib/scala-compiler-src.jar]
     [exec] Resolving [c4cd524dc29d298a5034637f6b31122dccb300d6/lib/scala-compiler.jar]
     [exec] Resolving [2cba5a13ef44bf93133be26cc89ba3a640a5c28f/lib/scala-library-src.jar]
     [exec] Resolving [d7c6e69eba3dba2f75a0f44e56480cd3dbab8931/lib/scala-library.jar]
     [exec] Resolving [dbc00bd4b09012aa52e802926dee3f8a02a767ff/lib/scala-reflect-src.jar]
     [exec] Resolving [c0eed5dee0a3204239c9b35134cad8b3ad140121/lib/scala-reflect.jar]
     [exec] Resolving [e737b123d31eede5594ceda07caafed1673ec472/test/files/codelib/code.jar]
     [exec] Resolving [02fe2ed93766323a13f22c7a7e2ecdcd84259b6c/test/files/lib/annotations.jar]
     [exec] Resolving [981392dbd1f727b152cd1c908c5fce60ad9d07f7/test/files/lib/enums.jar]
     [exec] Resolving [b1ec8a095cec4902b3609d74d274c04365c59c04/test/files/lib/genericNest.jar]
     [exec] Resolving [346d3dff4088839d6b4d163efa2892124039d216/test/files/lib/jsoup-1.3.1.jar]
     [exec] Resolving [be8454d5e7751b063ade201c225dcedefd252775/test/files/lib/methvsfield.jar]
     [exec] Resolving [cd33e0a0ea249eb42363a2f8ba531186345ff68c/test/files/lib/nest.jar]
     [exec] Resolving [b6f4dbb29f0c2ec1eba682414f60d52fea84f703/test/files/lib/scalacheck.jar]
     [exec] Resolving [1b11ac773055c1e942c6b5eb4aabdf02292a7194/test/files/speclib/instrumented.jar]
     [exec] Resolving [a1883f4304d5aa65e1f6ee6aad5900c62dd81079/tools/push.jar]

init.maven.tasks:

init.maven.jars:
[artifact:dependencies] There were no dependencies specified

init.hasbuildnum:

init.build.snapshot:

init.build.release:

init.build.nopatch.release:

init.build.patch.release:

init.hasmavensuffix:

init.osgi.suffix:

init.osgi.suffix.final:

init.osgi.suffix.snapshot:

init.build.suffix.done:

init.version.git:

init.version.release:

init.version.snapshot:

init.version.done:

init.testjava6:

init.fail.bad.jdk:

init.warn.jdk7:

init.extra.tasks:
[artifact:dependencies] [WARNING] POM for 'biz.aQute:bndlib:pom:1.43.0:compile' is invalid.
[artifact:dependencies]
[artifact:dependencies] Its dependencies (if any) will NOT be available to the current build.

init:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/lib/extra
     [echo]        build time: 14 March 2013, 16:41:35
     [echo]      java version: Java HotSpot(TM) 64-Bit Server VM 1.6.0_43 (1.6)
     [echo]         java args: -Xms1536M -Xmx3072M -Xss2M -XX:MaxPermSize=512M -XX:+UseParallelGC
     [echo]        javac args:
     [echo]       scalac args: -Yreify-copypaste
     [echo] scalac quick args: -Yreify-copypaste
     [echo]          git date: 20130314-075038
     [echo]          git hash: d229fb5273
     [echo]     maven version: 2.11.0-SNAPSHOT
     [echo]      OSGi version: 2.11.0.v20130314-075038-d229fb5273
     [echo] canonical version: 2.11.0-20130314-075038-d229fb5273

asm.clean:
   [delete] Deleting directory /Users/xeno_by/Projects/Kepler_7251/build/asm

libs.clean:
   [delete] Deleting directory /Users/xeno_by/Projects/Kepler_7251/build/libs

quick.clean:
   [delete] Deleting directory /Users/xeno_by/Projects/Kepler_7251/build/quick

palo.clean:

locker.clean:
   [delete] Deleting directory /Users/xeno_by/Projects/Kepler_7251/build/locker

docs.clean:

dist.clean:

sbt.clean:

osgi.clean:

all.clean:

init.jars.check:

init.jars:

init.maven.tasks:

init.maven.jars:
[artifact:dependencies] There were no dependencies specified

init.hasbuildnum:

init.build.snapshot:

init.build.release:

init.build.nopatch.release:

init.build.patch.release:

init.hasmavensuffix:

init.osgi.suffix:

init.osgi.suffix.final:

init.osgi.suffix.snapshot:

init.build.suffix.done:

init.version.git:

init.version.release:

init.version.snapshot:

init.version.done:

init.testjava6:

init.fail.bad.jdk:

init.warn.jdk7:

init.extra.tasks:
[artifact:dependencies] [WARNING] POM for 'biz.aQute:bndlib:pom:1.43.0:compile' is invalid.
[artifact:dependencies]
[artifact:dependencies] Its dependencies (if any) will NOT be available to the current build.

init:
     [echo]        build time: 14 March 2013, 16:41:35
     [echo]      java version: Java HotSpot(TM) 64-Bit Server VM 1.6.0_43 (1.6)
     [echo]         java args: -Xms1536M -Xmx3072M -Xss2M -XX:MaxPermSize=512M -XX:+UseParallelGC
     [echo]        javac args:
     [echo]       scalac args: -Yreify-copypaste
     [echo] scalac quick args: -Yreify-copypaste
     [echo]          git date: 20130314-075038
     [echo]          git hash: d229fb5273
     [echo]     maven version: 2.11.0-SNAPSHOT
     [echo]      OSGi version: 2.11.0.v20130314-075038-d229fb5273
     [echo] canonical version: 2.11.0-20130314-075038-d229fb5273

asm.start:

asm.lib:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/asm/classes
    [javac] Compiling 77 source files to /Users/xeno_by/Projects/Kepler_7251/build/asm/classes
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
[stopwatch] [asm.lib.timer: 1.436 sec]

asm.done:

forkjoin.start:

forkjoin.lib:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/libs/classes/forkjoin
    [javac] Compiling 10 source files to /Users/xeno_by/Projects/Kepler_7251/build/libs/classes/forkjoin
    [javac] Creating empty /Users/xeno_by/Projects/Kepler_7251/build/libs/classes/forkjoin/scala/concurrent/forkjoin/package-info.class
[stopwatch] [forkjoin.lib.timer: 1.417 sec]

forkjoin.pack:
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/libs/forkjoin.jar

forkjoin.done:

locker.start:

locker.pre-lib:

locker.lib:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/library
    [javac] Compiling 32 source files to /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/library
    [javac] Note: /Users/xeno_by/Projects/Kepler_7251/src/library/scala/collection/concurrent/MainNode.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
[scalacfork] Compiling 704 files to /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/library
[scalacfork] /Users/xeno_by/Projects/Kepler_7251/src/library/scala/SerialVersionUID.scala:15: warning: Implementation restriction: subclassing Classfile does not
[scalacfork] make your annotation visible at runtime.  If that is what
[scalacfork] you want, you must write the annotation class in Java.
[scalacfork] class SerialVersionUID(value: Long) extends scala.annotation.ClassfileAnnotation
[scalacfork]       ^
[scalacfork] warning: there were 72 deprecation warnings; re-run with -deprecation for details
[scalacfork] warning: there were 5 feature warnings; re-run with -feature for details
[scalacfork] three warnings found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/library/library.properties
[stopwatch] [locker.lib.timer: 1:07.961 sec]

locker.pre-reflect:

locker.reflect:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/reflect
[scalacfork] Compiling 137 files to /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/reflect
[scalacfork] warning: there were 25 deprecation warnings; re-run with -deprecation for details
[scalacfork] warning: there were 1 unchecked warnings; re-run with -unchecked for details
[scalacfork] warning: there were 6 feature warnings; re-run with -feature for details
[scalacfork] three warnings found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/reflect/reflect.properties
[stopwatch] [locker.reflect.timer: 44.847 sec]

locker.pre-comp:

locker.comp:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/compiler
[scalacfork] Compiling 275 files to /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/compiler
[scalacfork] /Users/xeno_by/Projects/Kepler_7251/src/compiler/scala/tools/nsc/util/AbstractFileClassLoader.scala:11: warning: imported `ScalaClassLoader' is permanently hidden by definition of object ScalaClassLoader in package util
[scalacfork] import util.ScalaClassLoader
[scalacfork]             ^
[scalacfork] warning: there were 32 deprecation warnings; re-run with -deprecation for details
[scalacfork] warning: there were 4 unchecked warnings; re-run with -unchecked for details
[scalacfork] warning: there were 13 feature warnings; re-run with -feature for details
[scalacfork] four warnings found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/compiler/compiler.properties
     [copy] Copying 4 files to /Users/xeno_by/Projects/Kepler_7251/build/locker/classes/compiler
[stopwatch] [locker.comp.timer: 1:15.355 sec]

locker.done:

quick.start:

quick.pre-lib:

quick.lib:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library
    [javac] Compiling 32 source files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library
    [javac] Note: /Users/xeno_by/Projects/Kepler_7251/src/library/scala/collection/concurrent/MainNode.java uses unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] Compiling 41 source files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
[scalacfork] Compiling 704 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library
[scalacfork] /Users/xeno_by/Projects/Kepler_7251/src/library/scala/SerialVersionUID.scala:15: warning: Implementation restriction: subclassing Classfile does not
[scalacfork] make your annotation visible at runtime.  If that is what
[scalacfork] you want, you must write the annotation class in Java.
[scalacfork] class SerialVersionUID(value: Long) extends scala.annotation.ClassfileAnnotation
[scalacfork]       ^
[scalacfork] warning: there were 73 deprecation warning(s); re-run with -deprecation for details
[scalacfork] warning: there were 1 unchecked warning(s); re-run with -unchecked for details
[scalacfork] warning: there were 5 feature warning(s); re-run with -feature for details
[scalacfork] four warnings found
[scalacfork] Compiling 49 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library
[scalacfork] warning: there were 216 deprecation warning(s); re-run with -deprecation for details
[scalacfork] one warning found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library/library.properties
     [copy] Copying 1 file to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/library
[stopwatch] [quick.lib.timer: 1:21.841 sec]

quick.pre-reflect:

quick.reflect:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/reflect
[scalacfork] Compiling 137 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/reflect
[scalacfork] warning: there were 25 deprecation warning(s); re-run with -deprecation for details
[scalacfork] warning: there were 6 feature warning(s); re-run with -feature for details
[scalacfork] two warnings found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/reflect/reflect.properties
[stopwatch] [quick.reflect.timer: 44.620 sec]

quick.pre-comp:

quick.comp:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/compiler
[scalacfork] Compiling 275 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/compiler
[scalacfork] /Users/xeno_by/Projects/Kepler_7251/src/compiler/scala/tools/nsc/util/AbstractFileClassLoader.scala:11: warning: imported `ScalaClassLoader' is permanently hidden by definition of object ScalaClassLoader in package util
[scalacfork] import util.ScalaClassLoader
[scalacfork]             ^
[scalacfork] warning: there were 32 deprecation warning(s); re-run with -deprecation for details
[scalacfork] warning: there were 8 unchecked warning(s); re-run with -unchecked for details
[scalacfork] warning: there were 13 feature warning(s); re-run with -feature for details
[scalacfork] four warnings found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/compiler/compiler.properties
     [copy] Copying 5 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/compiler
[stopwatch] [quick.comp.timer: 1:15.484 sec]

quick.pre-repl:

quick.repl:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/repl
[scalacfork] Compiling 48 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/repl
[scalacfork] // produced from source-/Users/xeno_by/Projects/Kepler_7251/src/repl/scala/tools/nsc/interpreter/IMain.scala,line-557,offset=22030
[scalacfork] object Test extends App {
[scalacfork]   val $u: scala.reflect.runtime.`package`.universe.type = scala.reflect.runtime.`package`.universe;
[scalacfork]   val $m: $u.Mirror = scala.reflect.runtime.`package`.universe.runtimeMirror(Predef.this.classOf[scala.tools.nsc.interpreter.IMain].getClassLoader);
[scalacfork]   import $u._, $m._
[scalacfork]   val tpe = staticClass("javax.script.Bindings").asType.toTypeConstructor
[scalacfork]   println(tpe)
[scalacfork] }
[scalacfork] warning: there were 2 feature warning(s); re-run with -feature for details
[scalacfork] one warning found

quick.pre-plugins:

quick.plugins:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/continuations-plugin
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/continuations-library
[scalacfork] Compiling 5 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/continuations-plugin
     [copy] Copying 1 file to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/continuations-plugin
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/misc/scala-devel/plugins
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/quick/misc/scala-devel/plugins/continuations.jar
[scalacfork] Compiling 2 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/continuations-library
[stopwatch] [quick.plugins.timer: 16.738 sec]

quick.pre-scalacheck:

quick.scalacheck:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scalacheck
[scalacfork] Compiling 14 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scalacheck
[stopwatch] [quick.scalacheck.timer: 19.309 sec]

quick.pre-scalap:

quick.scalap:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scalap
[scalacfork] Compiling 28 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scalap
[stopwatch] [quick.scalap.timer: 19.742 sec]

quick.pre-partest:

quick.partest:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/partest
    [javac] Compiling 6 source files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/partest
[scalacfork] Compiling 31 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/partest
[scalacfork] warning: there were 5 deprecation warning(s); re-run with -deprecation for details
[scalacfork] one warning found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/partest/partest.properties
     [copy] Copying 1 file to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/partest
[stopwatch] [quick.partest.timer: 19.041 sec]

quick.pre-scaladoc:

quick.scaladoc:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scaladoc
[scalacfork] Compiling 48 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scaladoc
[scalacfork] warning: there were 5 deprecation warning(s); re-run with -deprecation for details
[scalacfork] one warning found
[propertyfile] Creating new property file: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scaladoc/scaladoc.properties
     [copy] Copying 67 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/scaladoc

quick.pre-interactive:

quick.interactive:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/interactive
[scalacfork] Compiling 25 files to /Users/xeno_by/Projects/Kepler_7251/build/quick/classes/interactive
[scalacfork] /Users/xeno_by/Projects/Kepler_7251/src/interactive/scala/tools/nsc/interactive/RangePositions.scala:11: warning: Implementation restriction: early definitions in traits are not initialized before the super class is initialized.
[scalacfork]   override val useOffsetPositions = false
[scalacfork]                ^
[scalacfork] warning: there were 3 deprecation warning(s); re-run with -deprecation for details
[scalacfork] two warnings found

quick.pre-bin:

quick.bin:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/quick/bin

quick.done:

pack.start:

pack.pre-lib:

pack.lib:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-library.jar
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-actors.jar

pack.swing:
      [jar] Building MANIFEST-only jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-swing.jar

pack.pre-reflect:

pack.reflect:
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-reflect.jar

pack.pre-comp:

pack.comp:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/pack/META-INF
     [copy] Copying 1 file to /Users/xeno_by/Projects/Kepler_7251/build/pack/META-INF
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-compiler.jar
     [copy] Copying 1 file to /Users/xeno_by/Projects/Kepler_7251/build/pack/lib

pack.pre-plugins:

pack.plugins:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/pack/misc/scala-devel/plugins
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/misc/scala-devel/plugins/continuations.jar

pack.scalacheck:
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scalacheck.jar

pack.pre-partest:

pack.partest:
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-partest.jar
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scala-partest-javaagent.jar

pack.pre-scalap:

pack.scalap:
      [jar] Building jar: /Users/xeno_by/Projects/Kepler_7251/build/pack/lib/scalap.jar

pack.pre-bin:

pack.bin:
    [mkdir] Created dir: /Users/xeno_by/Projects/Kepler_7251/build/pack/bin

pack.done:

build:

BUILD SUCCESSFUL
Total time: 9 minutes 16 seconds
```

**Comment by xeno-by** [comment]
I'll reformulate the question: "Does it fail on your machine?"

**Comment by paulp** [comment]
Sorry, I'm beset from all sides. No, it doesn't fail for me either. That's annoying.

**Comment by paulp** [comment]
@adriaanm Any ideas why jenkins is the only guy with an issue?

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
@xeno-by I always forget the same difference. It only fails with -optimise. That damages my programmer's intuition about who is involved so feel free to move on.

**Comment by paulp** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
This tests my theory that it's a bug in the inliner.

**Comment by paulp** [comment]
Going to reopen targeting 2.10.x.

**Comment by paulp** [comment]
@xeno-by And let's say I owe you dinner.

**Pull request closed without merging**